### ISSUE TITLE
[1.21 Fabric] Just Crashes

### ISSUE THREAD

**Issue by ToneyFoxxy** [jellysquid3/Lithium#555]
### Version: 
>lithium-fabric-mc1.21-0.13.0

### Reproduction Steps:
> 1. Launch the game with Lithium installed

### Crash Report File:
> [https://gist.github.com/ToneyFoxxy/52a8bf88a10ee361c69f70305dfcf52a#file-gistfile1-txt](https://gist.github.com/ToneyFoxxy/52a8bf88a10ee361c69f70305dfcf52a#file-gistfile1-txt)

### Additional Information:
>I'm running Artix Linux on the XanMod kernel
>Edit: Working currently on 1.20.2 (lithium-fabric-mc1.20.2-0.12.0)

**Comment by 2No2Name** [comment]
Workaround:
Add the line
`mixin.minimal_nonvanilla.world.block_entity_ticking.support_cache=false`
to the lithium.properties in the config folder in the minecraft folder.

**Comment by ToneyFoxxy** [comment]
> Workaround: Add the line `mixin.minimal_nonvanilla.world.block_entity_ticking.support_cache=false` to the lithium.properties in the config folder in the minecraft folder.

Still crashes
[https://gist.github.com/ToneyFoxxy/e3cff8f8372d69bb8c0599b3ec41835f#file-gistfile1-txt](https://gist.github.com/ToneyFoxxy/e3cff8f8372d69bb8c0599b3ec41835f#file-gistfile1-txt)

**Comment by ToneyFoxxy** [comment]
Also crashes on my brother's PC which is running Endeavour OS

**Comment by wilebaldo** [comment]
I have the same problem I'm playing on MacOS and with PrismLauncher, is crash with the lithium 0.13 for minecraft 1.21, but with the version of Lithium 0.12.7 the game is working without a problem.

this is my crash file.
[crash-2024-08-21_20.04.19-client.txt](https://github.com/user-attachments/files/16703522/crash-2024-08-21_20.04.19-client.txt)

**Comment by ClassyElm** [comment]
> > Workaround: Add the line `mixin.minimal_nonvanilla.world.block_entity_ticking.support_cache=false` to the lithium.properties in the config folder in the minecraft folder.
> 
> Still crashes https://gist.github.com/ToneyFoxxy/e3cff8f8372d69bb8c0599b3ec41835f#file-gistfile1-txt

@ToneyFoxxy, try also adding this line to the config:
`mixin.block.hopper=false`
https://github.com/CaffeineMC/lithium-fabric/blob/1.21/lithium-mixin-config.md#mixinblockhopper

**Comment by ToneyFoxxy** [comment]
@ClassyElm That one works on it's own, thanks

**Comment by zarunet** [comment]
Can confirm on 1.21.1-0.13.0. Same crashes with only Lithium installed. Had to disable both mixins mentioned above.

Running Ubuntu 24.04 on the default kernel, if that helps.

**Comment by 2No2Name** [comment]
I think some users had this issue when changing their game version without creating a new profile in their custom minecraft launcher. It is currently unclear why this issue happens

**Comment by zarunet** [comment]
I _did_ upgrade Minecraft in-place from 1.21.0 to 1.21.1. Will try to recreate a profile when i get back to my computer next wednesday

**Comment by Earthcomputer** [comment]
Ideally when you do something like that it's good to check it works on the officially supported version before creating an issue on GitHub.

**Comment by 2No2Name** [comment]
I don't think it is expected that changing the version in a minecraft launcher behaves differently from creating an equivalent profile with the new version

**Comment by idscic** [comment]
i 

> I think some users had this issue when changing their game version without creating a new profile in their custom minecraft launcher. It is currently unclear why this issue happens

seconded, updating in place instead of new profile causes crash however fresh instance does not cause a game crash

**Comment by willrun4fun** [comment]
Fixed my issue as well.  Works on fresh profile.
It wasn't to much hassle to pull over my xaeros maps and a few other UI mod configs and I'm good now.

**Comment by MrPacha23** [comment]
Hi, I'm trying to use it on Windows 11 and it crashes, I tried the lines you sent previously but it didn't work

[crash-2024-09-14_11.07.41-client.txt](https://github.com/user-attachments/files/17003244/crash-2024-09-14_11.07.41-client.txt)

**Comment by deiphiz** [comment]
Chiming in here as a Prism Launcher user. I also had this same issue at first, and upon hearing people had it fixed by creating a new profile/instance it led me to see if there's some sort of cache or old option I had to delete. Turns out there was.

Just got to delete the .minecraft/.fabric folder in the instance. The launcher will rebuild all the mappings and whatnot on launch and it should start working just like a fresh instance.

**Comment by MrPacha23** [comment]
> Chiming in here as a Prism Launcher user. I also had this same issue at first, and upon hearing people had it fixed by creating a new profile/instance it led me to see if there's some sort of cache or old option I had to delete. Turns out there was.
> 
> Just got to delete the .minecraft/.fabric folder in the instance. The launcher will rebuild all the mappings and whatnot on launch and it should start working just like a fresh instance.

That didn't work for me either, I play on Mojang launcher

**Comment by deiphiz** [comment]
> > Chiming in here as a Prism Launcher user. I also had this same issue at first, and upon hearing people had it fixed by creating a new profile/instance it led me to see if there's some sort of cache or old option I had to delete. Turns out there was.
> > Just got to delete the .minecraft/.fabric folder in the instance. The launcher will rebuild all the mappings and whatnot on launch and it should start working just like a fresh instance.
> 
> That didn't work for me either, I play on Mojang launcher

The error in your crash log is related to carpet, not Lithium.

**Comment by MrPacha23** [comment]
> > > Chiming in here as a Prism Launcher user. I also had this same issue at first, and upon hearing people had it fixed by creating a new profile/instance it led me to see if there's some sort of cache or old option I had to delete. Turns out there was.
> > > Just got to delete the .minecraft/.fabric folder in the instance. The launcher will rebuild all the mappings and whatnot on launch and it should start working just like a fresh instance.
> > 
> > 
> > That didn't work for me either, I play on Mojang launcher
> 
> The error in your crash log is related to carpet, not Lithium.

Is it known if there is any incompatibility between these mods?

**Comment by zarunet** [comment]
> Chiming in here as a Prism Launcher user. I also had this same issue at first, and upon hearing people had it fixed by creating a new profile/instance it led me to see if there's some sort of cache or old option I had to delete. Turns out there was.
> 
> Just got to delete the .minecraft/.fabric folder in the instance. The launcher will rebuild all the mappings and whatnot on launch and it should start working just like a fresh instance.

Can confirm this worked for me! It thus appears that the way forward would be for fabric to purge this cache whenever its version changes. This would probably be better suited as a feature request on `fabric-launcher`'s side tho.

**Comment by owowed** [comment]
I may found a fix while upgrading my Prism Launcher instance to 1.21.1.

The previous solution deleting `.minecraft/.fabric` did not work for me, so I had to search for other things in my instance.

Turns out, when you change your Minecraft version to 1.21.1, the launcher does not automatically change the "Intermediary Mappings" version to 1.21.1. So I changed that too, and voila, my Minecraft 1.21.1 launched just fine with Lithium.
